task/sql-tasks.js

```diff
@@ -280,7 +463,25 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+    SELECT DISTINCT
+        Customers.CompanyName AS "CompanyName",
+        Products.ProductName AS "ProductName",
+        OrderDetails.UnitPrice AS "PricePerItem"
+    FROM Customers
+    INNER JOIN Orders ON Customers.CustomerID = Orders.CustomerID 
+    INNER JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID
+    INNER JOIN Products ON OrderDetails.ProductID = Products.ProductID
+    WHERE OrderDetails.UnitPrice = (SELECT MAX(OrderDetails2.UnitPrice)
+        FROM Customers AS Customers2
+        INNER JOIN Orders AS Orders2 ON Customers2.CustomerID = Orders2.CustomerID 
+        INNER JOIN OrderDetails AS OrderDetails2 ON Orders2.OrderID = OrderDetails2.OrderID 
+        WHERE Customers2.CompanyName = Customers.CompanyName) 

```

Comment:
companyname doesn't uniquely identify customer according to the schema. Also this column doesn't have an index and it makes the query slow